openpype/hosts/blender/plugins/publish/unpause_sync_server.py

```diff
@@ -17,9 +93,37 @@ def process(self, context):
         sync_server_module = manager.modules_by_name["sync_server"]
         sync_server_module.unpause_server()
 
+        # Compile match patterns
+        match_tb = re.compile(
+            "|".join(f"Traceback.*{e}: .*?\n" for e in BUILTIN_EXCEPTIONS),
+            re.DOTALL,
+        )
+        match_blend_file = re.compile("Read blend:(.*)")
+
         # Wait for all started futures to finish
+        subprocess_errors = False
         for instance in context:
-            for future in instance.data.get("representations_futures", []):
-                while future.running():
-                    self.log.debug(f"Waiting for {future} to finish...")
-                    sleep(1)
+            for future in as_completed(
+                instance.data.get("representations_futures", [])
+            ):
+                result = future.result().decode()
+
+                # Iterate through matched errors
+                if errors_stack := re.finditer(match_tb, result):
+                    # Match file path
+                    blend_file = re.search(match_blend_file, result)
+
+                    # Notify matched erros in log
+                    for stack in errors_stack:
+                        self.log.error(
+                            f"Blend file: {blend_file[1]}\n\n{stack[0]}"
+                        )
+                    subprocess_errors = True
+                else:
+                    self.log.info(result)
+
+        # Stop if errors
+        if subprocess_errors:
+            raise RuntimeError(
+                "Errors happened during subprocesses. See above."
```

Comment:
```suggestion
                "Errors occured during subprocesses. See above."
```
I know, I'm picky, but it sounds better that way :ok_hand: